src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,248 @@
+
+    _isOngoingTopicCreationDone = true;
+    _adminClient = adminClient;
+    _executor = Executors.newSingleThreadScheduledExecutor();
+    _reportIntervalSecond = Duration.ofSeconds(1);
+    _running = new AtomicBoolean(false);
+    _configDefinedServiceName = name;
+    // TODO: instantiate a new instance of ClusterTopicManipulationMetrics(..) here.
+  }
+
+  /**
+   * The start logic must only execute once.  If an error occurs then the implementer of this class must assume that
+   * stop() will be called to clean up.  This method must be thread safe and must assume that stop() may be called
+   * concurrently. This can happen if the monitoring application's life cycle is being managed by a container.  Start
+   * will only be called once.
+   */
+  @Override
+  public void start() {
+    if (_running.compareAndSet(false, true)) {
+      LOGGER.info("ClusterTopicManipulationService started for {} - {}", _configDefinedServiceName,
+          this.getClass().getCanonicalName());
+      Runnable clusterTopicManipulationServiceRunnable = new ClusterTopicManipulationServiceRunnable();
+
+      _executor.scheduleAtFixedRate(clusterTopicManipulationServiceRunnable, _reportIntervalSecond.getSeconds(),
+          _reportIntervalSecond.getSeconds(), TimeUnit.SECONDS);
+    }
+  }
+
+  private class ClusterTopicManipulationServiceRunnable implements Runnable {
+
+    private ClusterTopicManipulationServiceRunnable() {
+      // unaccessed.
+    }
+
+    /**
+     * When an object implementing interface <code>Runnable</code> is used
+     * to create a thread, starting the thread causes the object's
+     * <code>run</code> method to be called in that separately executing
+     * thread.
+     * <p>
+     * The general contract of the method <code>run</code> is that it may
+     * take any action whatsoever.
+     *
+     * @see     Thread#run()
+     */
+    @Override
+    public void run() {
+      try {
+        ClusterTopicManipulationService.this.createDeleteClusterTopic();
+      } catch (Exception e) {
+        LOGGER.error("{} {} failed to run createDeleteClusterTopic()", _configDefinedServiceName,
+            ClusterTopicManipulationService.this.getClass().getSimpleName(), e);
+      }
+    }
+  }
+
+  private void createDeleteClusterTopic() {
+
+    if (_isOngoingTopicCreationDone) {
+
+      int random = ThreadLocalRandom.current().nextInt();
+      _currentlyOngoingTopic = XinfraMonitorConstants.TOPIC_MANIPULATION_SERVICE_TOPIC + Math.abs(random);
+
+      try {
+        int brokerCount = _adminClient.describeCluster().nodes().get().size();
+        _createTopicsResult = _adminClient.createTopics(Collections.singleton(
+            new NewTopic(_currentlyOngoingTopic, XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS,
+                (short) brokerCount)));
+
+        _totalPartitions.set(brokerCount * XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS);
```

Comment:
I actually don't understand this line. I thought `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS ` is the number of partitions in the created topic. But `brokerCount * XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` seems like we are expecting `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` partitions on every broker???

Reply:
`brokerCount * XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS)` is the total number of replicas (partitions) that exist for a topic. 
`TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` is the number of partitions (not the RF)

Reply:
If `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` is the number of partition for a topic, shouldn't the total number of replicas that exist for a topic be `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` * RF instead of `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` * brokerCount?

`TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS * brokerCount` seems that each broker has `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS` partition.

I am confused about how is the total number of replicas associated with the broker count.

Reply:
The RF is set to the brokerCount currently to enable maximize spreading the partitions and replicas across all the brokers in the clusters as possible.
Feel free to disagree or suggest otherwise.

Reply:
I think the below statement is totally valid and reasonable.
```
The RF is set to the brokerCount currently to enable maximize assigning the many partitions and replicas across all the brokers in the clusters as possible
```
Maybe we should document or annotate the code with this information? Maybe this information is documented somewhere in this codebase. But just looking at this part, I had a difficult time figuring out why `TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS * brokerCount`

Reply:
Added. documentation on it.